File: frontend/src/types/appointment.ts
export interface BookingRequest {
  mentorId: string;
  startAt: string;
  endAt: string;
}

// ===== Booking DTOs (mentor side) =====

export type AppointmentStatusEnum =
  | "Pending"
  | "Confirmed"
  | "Cancelled"
  | "Completed";

export interface MenteeInfoDto {
  name: string;
  email: string;
  avatar?: string | null;
}

export interface MentorAppointmentDto {
  mentorId: string; // Guid
  mentee: MenteeInfoDto;
  startAt: string; // ISO-8601 UTC string
  endAt: string; // ISO-8601 UTC string
  status: AppointmentStatusEnum;
  meetingLink?: string | null;
}

export interface MentorInfoDto {
  name: string;
  email: string;
  avatar?: string | null;
  company?: string | null;
  position?: string | null;
}

export interface MenteeAppointmentDto {
  mentorId: string; // Guid → string
  mentor: MentorInfoDto;
  startAt: string; // DateTime → ISO-8601 UTC string
  endAt: string;
  status: AppointmentStatusEnum;
  meetingLink?: string | null;
}

File: backend/Models/Appointment.cs
namespace backend.Models;

public class Appointment
{
    public Guid Id { get; set; }

    public Guid MentorId { get; set; }

    public Guid MenteeId { get; set; }

    /// <summary>
    /// Stored in UTC
    /// </summary>
    public DateTime StartAt { get; set; }

    /// <summary>
    /// Stored in UTC
    /// </summary>
    public DateTime EndAt { get; set; }

    /// <summary>
    /// Pending | Confirmed | Cancelled | Completed
    /// </summary>
    public AppointmentStatusEnum Status { get; set; }

    public string? MeetingLink { get; set; }
    
    public string? GoogleCalendarLink { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User Mentor { get; set; }
    public User Mentee { get; set; }
    public virtual ICollection<MentorReview> Reviews { get; set; } = new List<MentorReview>();
}
